webserver/views/api/v1/datasets.py

```diff
@@ -136,18 +137,47 @@ def add_class(dataset_id):
         {
             "name": "Not Mood",
             "description": "Dataset for mood misclassification.",
+            "recordings": ["770cc467-8dde-4d22-bc4c-a42f91e"]
         }
 
     :reqheader Content-Type: *application/json*
     :<json string name: *Required.* Name of the class. Must be unique within a dataset.
     :<json string description: *Optional.* Description of the class.
-    :<json array public: *Optional.* Array of recording MBIDs (``string``) to add into that class. For example:
+    :<json array recordings: *Optional.* Array of recording MBIDs (``string``) to add into that class. For example:
         ``["770cc467-8dde-4d22-bc4c-a42f91e"]``.
 
 
     :resheader Content-Type: *application/json*
     """
-    raise NotImplementedError
+    ds = get_check_dataset(dataset_id)
+    if ds["author"] != current_user.id:
```

Comment:
Right, we could have 

    def get_check_dataset(dataset_id, write=False)

with the default we don't need to change existing read-only uses of the function, and we can use `write=True` for the API edit endpoints